revvy/robot/remote_controller.py

```diff
@@ -12,10 +13,84 @@
 RemoteControllerCommand = namedtuple('RemoteControllerCommand', ['analog', 'buttons', 'background_command'])
 
 
+class AutonomousModeRequest:
+    NONE   = 0
+    STOP   = 1
+    START  = 2
+    PAUSE  = 3
+    RESUME = 4
+
+    def __init__(self):
+        self.__state = AutonomousModeRequest.NONE
+
+    def set_start_request(self):
+        self.__state = AutonomousModeRequest.START
+
+    def set_stop_request(self):
+        self.__state = AutonomousModeRequest.STOP
+
+    def set_pause_request(self):
+        self.__state = AutonomousModeRequest.PAUSE
+
+    def set_resume_request(self):
+        self.__state = AutonomousModeRequest.RESUME
+
+    def is_start_pending(self):
+        return self.__state == AutonomousModeRequest.START
+
+    def is_stop_pending(self):
+        return self.__state == AutonomousModeRequest.STOP
+
+    def is_pause_pending(self):
+        return self.__state == AutonomousModeRequest.PAUSE
+
+    def is_resume_pending(self):
+        return self.__state == AutonomousModeRequest.RESUME
+
+    def clear_pending(self):
+        self.__state = AutonomousModeRequest.NONE
+
+
+class BackgroundControlState:
+    STOPPED = 1
+    RUNNING = 2
+    PAUSED  = 3
+
+    def __init__(self):
```

Comment:
Same as above, really. These classes should be enums if enums exist on the robot's python.